### ISSUE TITLE
Inquirer.prompt().ui is hidden

### ISSUE THREAD

**Issue by MarcoScabbiolo** [yeoman/generator#1073]
https://github.com/yeoman/generator/blob/b4f1bfcc5466e3aed004cff8c86e5bf5306195e6/lib/index.js#L155

Using `then` after the prompt resolves hides the `ui` property of inquirer which allows to subscribe to `process` to use `rxjs` for more complex prompting flows.

Maybe having a new "observePrompts" method that takes a Subject and returns an Observable could do the trick, but it would also be necessary to have new storing and prefilling methods for a single question/answer: `promptSuggestion.storeAnswer` and `promptSuggestion.prefillQuestion`.

**Comment by homj** [comment]
@SBoudrias the docs you linked contain the following statement (at [Adapter#prompt()](https://yeoman.io/authoring/integrating-yeoman.html#adapterprompt)):

> Its signature and behavior follows these of Inquirer.js

According to this statement, a call to `Generator#prompt()` should indeed accept Observables and return the `ui` property of `inquirer#prompt()`, since that is part of its method signature as described in [https://github.com/SBoudrias/Inquirer.js#reactive-interface](https://github.com/SBoudrias/Inquirer.js#reactive-interface)

As @MarcoScabbiolo stated, this would allow us to create more sophisticated dialogs between the user and the generator. However, if you still intend to not support this feature, the documentation should be updated to inform the readers that the signatures of `inquirer#prompt()` and `Generator#prompt()` don't match exactly.

**Comment by SBoudrias** [comment]
Yeah, this is confusing in the documentation. We should update to only state the arguments match.
